datanode/gateway/graphql/schema.graphql

```diff
@@ -7839,6 +7847,35 @@ enum AMMStatusReason {
   STATUS_REASON_CANNOT_REBASE
 }
 
+type NewProtocolAutomatedPurchase {
+  "The source token which will be swapped"
+  from: ID!
+  "The account type for the network from which the tokens will be sourced"
+  fromAccountType: AccountType!
+  "The account type for the network to which the purchased tokens will be sent"
+  toAccountType: AccountType!
+  "The market which will be used to enact the purchase/sale"
+  marketId: ID
+  "The oracle which will define an approximate acceptable price for the transaction"
+  priceOracle: DataSourceDefinition!
+  "The final acceptable price is the price oracle value weighted by this factor (e.g. 1.05 to allow for 5% slippage on the purchase)"
+  oracleOffsetFactor: String! 
+  "A time based oracle for when auctions will occur"
+  auctionSchedule: DataSourceDefinition
+  "The duration of the auction"
+  auction_duration: String!
```

Comment:
Should this be formatted auctionDuration?

Reply:
yes, I'll fix